includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -497,6 +497,11 @@ public function is_in_test_mode() {
 	 * @return bool Whether the gateway is enabled and ready to accept payments.
 	 */
 	public function is_available() {
+		// Disable the gateway if using live mode without HTTPS set up.
+		if ( ! $this->is_in_test_mode() && ( ! wc_site_is_https() || 'yes' !== get_option( 'woocommerce_force_ssl_checkout' ) ) ) {
```

Comment:
There's a [`wc_checkout_is_https` function](https://github.com/woocommerce/woocommerce/blob/b19500728b4b292562afb65eb3a0c0f50d5859de/includes/wc-conditional-functions.php#L408), which looks similar to this but includes a couple of more checks. Would it make sense to use it here?

Reply:
This is GOLD 🥇 Thanks for pointing it out, it's actually removing all the logic 👍